### ISSUE TITLE
Packer can't acquire IP from vSphere Cisco VM

### ISSUE THREAD

**Issue by bruteForceAttack** [hashicorp/packer#10363]
#### Overview of the Issue

Problem is with acquiring IP address from vmware vsphere (gathered from vm-tools) so Packer doesn't do the shutdown command via vmware tools. Information such as IP, hostname and so on however is present in vSphere also VM is reachable with SSH client from localhost.
So far this happens only with Cisco image of CSR 1000v: csr1000v-universalk9.16.09.01.iso other OS type like Linux or Windows is good.
Problem seems to be related to the change (GH-9450) [](https://github.com/hashicorp/packer/pull/9450) as per release notes for version number 1.6.1.
**Everything is fine** and works with previous versions e.g. **1.5.6** or **1.6.0**. I have tested 1.6.1+ and no success at all.

#### Reproduction Steps

`packer build -force -var-file=variables.json template-cisco-csr.json`

### Packer version

1.6.5 (1.6.1+)

### Simplified Packer Buildfile

Packer json template file:
```yaml
{
    "builders": [
      {
        "type": "vsphere-iso",
        "vcenter_server": "{{user `vcenter_server`}}",
        "username": "{{user `username`}}",
        "password": "{{user `password_vsphere`}}",
        "insecure_connection": "true",
        "vm_name": "{{user `template_prefix`}}_cisco_csr_16.9.1",
        "datacenter": "{{user `datacenter`}}",
        "cluster": "{{user `cluster`}}",
        "resource_pool": "{{user `resource_pool`}}",
        "datastore": "{{user `datastore`}}",
        "host": "{{user `host`}}",
        "folder": "{{user `folder`}}",
        "guest_os_type": "other3xLinux64Guest",
        "CPUs": 2,
        "RAM": 4096,
        "RAM_reserve_all": true,
        "disk_controller_type": "pvscsi",
        "storage": [
          {
            "disk_size": 8192,
            "disk_thin_provisioned": true
          }
        ],
        "network_adapters": [
          {
            "network": "2000",
            "network_card": "vmxnet3"
          }
        ],
        "iso_paths": [
          "[Install] Cisco/CSR1000v/csr1000v-universalk9.16.09.01.iso"
        ],
        "iso_checksum": "sha512:c578a59272f1f80bc044bc63d8bd7dd106cda7ccc419457cdfcc5f3984c323bcb84da9e4928f3dc0a1d61093efcf546ab51305918d435ad01720332550c2db7b",
        "convert_to_template": "false",
        "communicator": "ssh",
        "ssh_username": "admin",
        "ssh_password": "{{user `password`}}",
        "ip_settle_timeout": "60s",
        "boot_wait": "5s",
        "boot_command": [
          "<wait10><enter><wait10><wait10><wait10><wait10><wait10>",
          "<wait10><wait10><wait10><wait10><wait10><wait10>",
          "<wait10><wait10><wait10><wait10><wait10><wait10>",
          "<wait10><wait10><wait10><wait10><wait10><wait10>",
          "<wait10><wait10><wait10><wait10><wait10><wait10>",
          "<wait10>",
          "<enter>enable<enter>conf t<enter>int giga1<enter>ip addr dhcp<enter><wait5>exit<enter>",
          "enable secret password 0 {{user `password`}}<enter>no service config<enter>hostname router.cisco<enter>ip domain-name lan.net<enter>crypto key gen rsa mod 4096<enter><wait5>",
          "username admin password 0 {{user `password`}}<enter>line vty 0 15<enter>login local<enter>transport input ssh<enter>exit<enter>",
          "do wr<enter>end<enter>exit<enter>"
        ]
      }
    ]
  }
  
```

### Operating system and Environment details

```
uname -a
Darwin 20.1.0 Darwin Kernel Version 20.1.0: Sat Oct 31 00:07:11 PDT 2020; root:xnu-7195.50.7~2/RELEASE_X86_64 x86_64 i386 MacBookPro15,1 Darwin
```

### Log Fragments and crash.log files

No crash logs are generated. The build timeouts because of no SSH connection into VM.

Packers standard output:

```
vsphere-iso: output will be in this color.
==> vsphere-iso: Creating VM...
==> vsphere-iso: Customizing hardware...
==> vsphere-iso: Mounting ISO images...
==> vsphere-iso: Adding configuration parameters...
==> vsphere-iso: Set boot order temporary...
==> vsphere-iso: Power on VM...
==> vsphere-iso: Waiting 5s for boot...
==> vsphere-iso: Typing boot command...
==> vsphere-iso: Waiting for IP...
==> vsphere-iso: Timeout waiting for IP.
==> vsphere-iso: Clear boot order...
==> vsphere-iso: Power off VM...
==> vsphere-iso: Destroying VM...
Build 'vsphere-iso' errored after 36 minutes 13 seconds: Timeout waiting for IP.

==> Wait completed after 36 minutes 13 seconds

==> Some builds didn't complete successfully and had errors:
--> vsphere-iso: Timeout waiting for IP.

==> Builds finished but no artifacts were created.
```

Log file generated by packer (PACKER_LOG=1):
<details>
 <summary>log file</summary>

```
2020/12/09 15:18:14 [INFO] Packer version: 1.6.5 [go1.15.3 darwin amd64]
2020/12/09 15:18:14 Checking 'PACKER_CONFIG' for a config file path
2020/12/09 15:18:14 'PACKER_CONFIG' not set; checking the default config file path
2020/12/09 15:18:14 Attempting to open config file: .packerconfig
2020/12/09 15:18:14 [WARN] Config file doesn't exist: .packerconfig
2020/12/09 15:18:14 Setting cache directory: templates/packer/packer_cache
2020/12/09 15:18:14 Creating plugin client for path: templates/packer/packer
2020/12/09 15:18:14 Starting plugin: templates/packer/packer []string{"templates/packer/packer", "plugin", "packer-builder-vsphere-iso"}
2020/12/09 15:18:14 Waiting for RPC address for: templates/packer/packer
2020/12/09 15:18:14 packer plugin: [INFO] Packer version: 1.6.5 [go1.15.3 darwin amd64]
2020/12/09 15:18:14 packer plugin: Checking 'PACKER_CONFIG' for a config file path
2020/12/09 15:18:14 packer plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/12/09 15:18:14 packer plugin: Attempting to open config file: .packerconfig
2020/12/09 15:18:14 packer plugin: [WARN] Config file doesn't exist: .packerconfig
2020/12/09 15:18:14 packer plugin: Setting cache directory: templates/packer/packer_cache
2020/12/09 15:18:14 packer plugin: args: []string{"packer-builder-vsphere-iso"}
2020/12/09 15:18:15 packer plugin: Plugin address: unix /var/folders/3l/jj8w1c854ml0hfzcjgczq1w00000gn/T/packer-plugin283264697
2020/12/09 15:18:15 packer plugin: Waiting for connection...
2020/12/09 15:18:15 Received unix RPC address for templates/packer/packer: addr is /var/folders/3l/jj8w1c854ml0hfzcjgczq1w00000gn/T/packer-plugin283264697
2020/12/09 15:18:15 packer plugin: Serving a plugin connection...
2020/12/09 15:18:15 Preparing build: vsphere-iso
2020/12/09 15:18:15 ui: [1;32mvsphere-iso: output will be in this color.[0m
2020/12/09 15:18:15 ui: 
2020/12/09 15:18:15 Build debug mode: false
2020/12/09 15:18:15 Force build: true
2020/12/09 15:18:15 On error: 
2020/12/09 15:18:15 Waiting on builds to complete...
2020/12/09 15:18:15 Starting build run: vsphere-iso
2020/12/09 15:18:15 Running builder: vsphere-iso
2020/12/09 15:18:15 [INFO] (telemetry) Starting builder vsphere-iso
2020/12/09 15:18:15 packer plugin: No URLs were provided to Step Download. Continuing...
2020/12/09 15:18:15 packer plugin: No CD files specified. CD disk will not be made.
2020/12/09 15:18:15 ui: [1;32m==> vsphere-iso: Creating VM...[0m
2020/12/09 15:18:17 ui: [1;32m==> vsphere-iso: Customizing hardware...[0m
2020/12/09 15:18:17 ui: [1;32m==> vsphere-iso: Mounting ISO images...[0m
2020/12/09 15:18:17 packer plugin: Check if ISO path is a Content Library path
2020/12/09 15:18:18 packer plugin: ISO path not identified as a Content Library path
2020/12/09 15:18:18 packer plugin: Using [Install] Cisco/CSR1000v/csr1000v-universalk9.16.09.01.iso as the datastore path
2020/12/09 15:18:18 packer plugin: Creating CD-ROM on controller '&{{{} 200 0xc000cab720 <nil> <nil> <nil> 0 <nil>} 0 []}' with iso '[Install] Cisco/CSR1000v/csr1000v-universalk9.16.09.01.iso'
2020/12/09 15:18:21 ui: [1;32m==> vsphere-iso: Adding configuration parameters...[0m
2020/12/09 15:18:21 packer plugin: No floppy files specified. Floppy disk will not be made.
2020/12/09 15:18:21 ui: [1;32m==> vsphere-iso: Set boot order temporary...[0m
2020/12/09 15:18:21 ui: [1;32m==> vsphere-iso: Power on VM...[0m
2020/12/09 15:18:23 ui: [1;32m==> vsphere-iso: Waiting 5s for boot...[0m
2020/12/09 15:18:28 ui: [1;32m==> vsphere-iso: Typing boot command...[0m
2020/12/09 15:18:28 packer plugin: [INFO] Waiting 10s
2020/12/09 15:18:38 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:18:38 packer plugin: [INFO] Waiting 10s
2020/12/09 15:18:48 packer plugin: [INFO] Waiting 10s
2020/12/09 15:18:58 packer plugin: [INFO] Waiting 10s
2020/12/09 15:19:08 packer plugin: [INFO] Waiting 10s
2020/12/09 15:19:18 packer plugin: [INFO] Waiting 10s
2020/12/09 15:19:28 packer plugin: [INFO] Waiting 10s
2020/12/09 15:19:38 packer plugin: [INFO] Waiting 10s
2020/12/09 15:19:48 packer plugin: [INFO] Waiting 10s
2020/12/09 15:19:58 packer plugin: [INFO] Waiting 10s
2020/12/09 15:20:08 packer plugin: [INFO] Waiting 10s
2020/12/09 15:20:18 packer plugin: [INFO] Waiting 10s
2020/12/09 15:20:28 packer plugin: [INFO] Waiting 10s
2020/12/09 15:20:38 packer plugin: [INFO] Waiting 10s
2020/12/09 15:20:48 packer plugin: [INFO] Waiting 10s
2020/12/09 15:20:58 packer plugin: [INFO] Waiting 10s
2020/12/09 15:21:08 packer plugin: [INFO] Waiting 10s
2020/12/09 15:21:18 packer plugin: [INFO] Waiting 10s
2020/12/09 15:21:28 packer plugin: [INFO] Waiting 10s
2020/12/09 15:21:38 packer plugin: [INFO] Waiting 10s
2020/12/09 15:21:48 packer plugin: [INFO] Waiting 10s
2020/12/09 15:21:58 packer plugin: [INFO] Waiting 10s
2020/12/09 15:22:08 packer plugin: [INFO] Waiting 10s
2020/12/09 15:22:18 packer plugin: [INFO] Waiting 10s
2020/12/09 15:22:28 packer plugin: [INFO] Waiting 10s
2020/12/09 15:22:38 packer plugin: [INFO] Waiting 10s
2020/12/09 15:22:48 packer plugin: [INFO] Waiting 10s
2020/12/09 15:22:58 packer plugin: [INFO] Waiting 10s
2020/12/09 15:23:08 packer plugin: [INFO] Waiting 10s
2020/12/09 15:23:18 packer plugin: [INFO] Waiting 10s
2020/12/09 15:23:28 packer plugin: [INFO] Waiting 10s
2020/12/09 15:23:38 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:23:38 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:23:38 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:23:38 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:23:38 packer plugin: Sending char 'b', code CodeB, shift false
2020/12/09 15:23:38 packer plugin: Sending char 'l', code CodeL, shift false
2020/12/09 15:23:39 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:23:39 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:23:39 packer plugin: Sending char 'c', code CodeC, shift false
2020/12/09 15:23:39 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:23:39 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:23:39 packer plugin: Sending char 'f', code CodeF, shift false
2020/12/09 15:23:39 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:23:39 packer plugin: Sending char 't', code CodeT, shift false
2020/12/09 15:23:40 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:23:40 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:23:40 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:23:40 packer plugin: Sending char 't', code CodeT, shift false
2020/12/09 15:23:40 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:23:40 packer plugin: Sending char 'g', code CodeG, shift false
2020/12/09 15:23:40 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:23:40 packer plugin: Sending char 'g', code CodeG, shift false
2020/12/09 15:23:41 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:23:41 packer plugin: Sending char '1', code Code1, shift false
2020/12/09 15:23:41 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:23:41 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:23:41 packer plugin: Sending char 'p', code CodeP, shift false
2020/12/09 15:23:41 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:23:41 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:23:41 packer plugin: Sending char 'd', code CodeD, shift false
2020/12/09 15:23:42 packer plugin: Sending char 'd', code CodeD, shift false
2020/12/09 15:23:42 packer plugin: Sending char 'r', code CodeR, shift false
2020/12/09 15:23:42 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:23:42 packer plugin: Sending char 'd', code CodeD, shift false
2020/12/09 15:23:42 packer plugin: Sending char 'h', code CodeH, shift false
2020/12/09 15:23:42 packer plugin: Sending char 'c', code CodeC, shift false
2020/12/09 15:23:42 packer plugin: Sending char 'p', code CodeP, shift false
2020/12/09 15:23:43 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:23:43 packer plugin: [INFO] Waiting 5s
2020/12/09 15:23:48 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:23:48 packer plugin: Sending char 'x', code CodeX, shift false
2020/12/09 15:23:48 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:23:48 packer plugin: Sending char 't', code CodeT, shift false
2020/12/09 15:23:48 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:23:48 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:23:48 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:23:49 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:23:49 packer plugin: Sending char 'b', code CodeB, shift false
2020/12/09 15:23:49 packer plugin: Sending char 'l', code CodeL, shift false
2020/12/09 15:23:49 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:23:49 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:23:49 packer plugin: Sending char 's', code CodeS, shift false
2020/12/09 15:23:49 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:23:50 packer plugin: Sending char 'c', code CodeC, shift false
2020/12/09 15:23:50 packer plugin: Sending char 'r', code CodeR, shift false
2020/12/09 15:23:50 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:23:50 packer plugin: Sending char 't', code CodeT, shift false
2020/12/09 15:23:50 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:23:50 packer plugin: Sending char 'p', code CodeP, shift false
2020/12/09 15:23:50 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:23:50 packer plugin: Sending char 's', code CodeS, shift false
2020/12/09 15:23:51 packer plugin: Sending char 's', code CodeS, shift false
2020/12/09 15:23:51 packer plugin: Sending char 'w', code CodeW, shift false
2020/12/09 15:23:51 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:23:51 packer plugin: Sending char 'r', code CodeR, shift false
2020/12/09 15:23:51 packer plugin: Sending char 'd', code CodeD, shift false
2020/12/09 15:23:51 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:23:51 packer plugin: Sending char '0', code Code0, shift false
2020/12/09 15:23:51 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:23:52 packer plugin: Sending char 'X', code CodeX, shift false
2020/12/09 15:23:52 packer plugin: Sending char 'X', code CodeX, shift false
2020/12/09 15:23:52 packer plugin: Sending char 'X', code CodeX, shift true
2020/12/09 15:23:52 packer plugin: Sending char 'X', code CodeX, shift false
2020/12/09 15:23:52 packer plugin: Sending char 'X', code CodeX, shift false
2020/12/09 15:23:52 packer plugin: Sending char 'X', code CodeX, shift true
2020/12/09 15:23:52 packer plugin: Sending char 'X', code CodeX, shift true
2020/12/09 15:23:53 packer plugin: Sending char 'X', code CodeX, shift true
2020/12/09 15:23:53 packer plugin: Sending char 'X', code CodeX, shift false
2020/12/09 15:23:53 packer plugin: Sending char 'X', code CodeX, shift false
2020/12/09 15:23:53 packer plugin: Sending char 'X', code CodeX, shift true
2020/12/09 15:23:53 packer plugin: Sending char 'X', code CodeX, shift false
2020/12/09 15:23:53 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:23:53 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:23:53 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:23:54 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:23:54 packer plugin: Sending char 's', code CodeS, shift false
2020/12/09 15:23:54 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:23:54 packer plugin: Sending char 'r', code CodeR, shift false
2020/12/09 15:23:54 packer plugin: Sending char 'v', code CodeV, shift false
2020/12/09 15:23:54 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:23:54 packer plugin: Sending char 'c', code CodeC, shift false
2020/12/09 15:23:54 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:23:55 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:23:55 packer plugin: Sending char 'c', code CodeC, shift false
2020/12/09 15:23:55 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:23:55 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:23:55 packer plugin: Sending char 'f', code CodeF, shift false
2020/12/09 15:23:55 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:23:55 packer plugin: Sending char 'g', code CodeG, shift false
2020/12/09 15:23:56 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:23:56 packer plugin: Sending char 'h', code CodeH, shift false
2020/12/09 15:23:56 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:23:56 packer plugin: Sending char 's', code CodeS, shift false
2020/12/09 15:23:56 packer plugin: Sending char 't', code CodeT, shift false
2020/12/09 15:23:56 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:23:56 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:23:56 packer plugin: Sending char 'm', code CodeM, shift false
2020/12/09 15:23:57 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:23:57 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:23:57 packer plugin: Sending char 'r', code CodeR, shift false
2020/12/09 15:23:57 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:23:57 packer plugin: Sending char 'u', code CodeU, shift false
2020/12/09 15:23:57 packer plugin: Sending char 't', code CodeT, shift false
2020/12/09 15:23:57 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:23:57 packer plugin: Sending char 'r', code CodeR, shift false
2020/12/09 15:23:58 packer plugin: Sending char '.', code CodeFullStop, shift false
2020/12/09 15:23:58 packer plugin: Sending char 'c', code CodeC, shift false
2020/12/09 15:23:58 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:23:58 packer plugin: Sending char 's', code CodeS, shift false
2020/12/09 15:23:58 packer plugin: Sending char 'c', code CodeC, shift false
2020/12/09 15:23:58 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:23:58 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:23:59 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:23:59 packer plugin: Sending char 'p', code CodeP, shift false
2020/12/09 15:23:59 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:23:59 packer plugin: Sending char 'd', code CodeD, shift false
2020/12/09 15:23:59 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:23:59 packer plugin: Sending char 'm', code CodeM, shift false
2020/12/09 15:23:59 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:23:59 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:24:00 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:24:00 packer plugin: Sending char '-', code CodeHyphenMinus, shift false
2020/12/09 15:24:00 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:24:00 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:24:00 packer plugin: Sending char 'm', code CodeM, shift false
2020/12/09 15:24:00 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:24:00 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:01 packer plugin: Sending char 'l', code CodeL, shift false
2020/12/09 15:24:01 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:24:01 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:24:01 packer plugin: Sending char '.', code CodeFullStop, shift false
2020/12/09 15:24:01 packer plugin: Sending char 't', code CodeT, shift false
2020/12/09 15:24:01 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:24:01 packer plugin: Sending char 'c', code CodeC, shift false
2020/12/09 15:24:01 packer plugin: Sending char 'h', code CodeH, shift false
2020/12/09 15:24:02 packer plugin: Sending char '.', code CodeFullStop, shift false
2020/12/09 15:24:02 packer plugin: Sending char 'c', code CodeC, shift false
2020/12/09 15:24:02 packer plugin: Sending char 'c', code CodeC, shift false
2020/12/09 15:24:02 packer plugin: Sending char 'd', code CodeD, shift false
2020/12/09 15:24:02 packer plugin: Sending char 'c', code CodeC, shift false
2020/12/09 15:24:02 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:24:02 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:24:02 packer plugin: Sending char '.', code CodeFullStop, shift false
2020/12/09 15:24:03 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:24:03 packer plugin: Sending char 'r', code CodeR, shift false
2020/12/09 15:24:03 packer plugin: Sending char 'g', code CodeG, shift false
2020/12/09 15:24:03 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:24:03 packer plugin: Sending char 'c', code CodeC, shift false
2020/12/09 15:24:03 packer plugin: Sending char 'r', code CodeR, shift false
2020/12/09 15:24:03 packer plugin: Sending char 'y', code CodeY, shift false
2020/12/09 15:24:04 packer plugin: Sending char 'p', code CodeP, shift false
2020/12/09 15:24:04 packer plugin: Sending char 't', code CodeT, shift false
2020/12/09 15:24:04 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:24:04 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:04 packer plugin: Sending char 'k', code CodeK, shift false
2020/12/09 15:24:04 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:24:04 packer plugin: Sending char 'y', code CodeY, shift false
2020/12/09 15:24:04 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:05 packer plugin: Sending char 'g', code CodeG, shift false
2020/12/09 15:24:05 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:24:05 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:24:05 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:05 packer plugin: Sending char 'r', code CodeR, shift false
2020/12/09 15:24:05 packer plugin: Sending char 's', code CodeS, shift false
2020/12/09 15:24:05 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:24:05 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:06 packer plugin: Sending char 'm', code CodeM, shift false
2020/12/09 15:24:06 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:24:06 packer plugin: Sending char 'd', code CodeD, shift false
2020/12/09 15:24:06 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:06 packer plugin: Sending char '4', code Code4, shift false
2020/12/09 15:24:06 packer plugin: Sending char '0', code Code0, shift false
2020/12/09 15:24:06 packer plugin: Sending char '9', code Code9, shift false
2020/12/09 15:24:07 packer plugin: Sending char '6', code Code6, shift false
2020/12/09 15:24:07 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:24:07 packer plugin: [INFO] Waiting 5s
2020/12/09 15:24:12 packer plugin: Sending char 'u', code CodeU, shift false
2020/12/09 15:24:12 packer plugin: Sending char 's', code CodeS, shift false
2020/12/09 15:24:12 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:24:12 packer plugin: Sending char 'r', code CodeR, shift false
2020/12/09 15:24:12 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:24:12 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:24:13 packer plugin: Sending char 'm', code CodeM, shift false
2020/12/09 15:24:13 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:24:13 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:13 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:24:13 packer plugin: Sending char 'd', code CodeD, shift false
2020/12/09 15:24:13 packer plugin: Sending char 'm', code CodeM, shift false
2020/12/09 15:24:13 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:24:13 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:24:14 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:14 packer plugin: Sending char 'p', code CodeP, shift false
2020/12/09 15:24:14 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:24:14 packer plugin: Sending char 's', code CodeS, shift false
2020/12/09 15:24:14 packer plugin: Sending char 's', code CodeS, shift false
2020/12/09 15:24:14 packer plugin: Sending char 'w', code CodeW, shift false
2020/12/09 15:24:15 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:24:15 packer plugin: Sending char 'r', code CodeR, shift false
2020/12/09 15:24:15 packer plugin: Sending char 'd', code CodeD, shift false
2020/12/09 15:24:15 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:15 packer plugin: Sending char '0', code Code0, shift false
2020/12/09 15:24:15 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:16 packer plugin: Sending char 'X', code CodeX, shift false
2020/12/09 15:24:16 packer plugin: Sending char 'X', code CodeX, shift false
2020/12/09 15:24:16 packer plugin: Sending char 'X', code CodeX, shift true
2020/12/09 15:24:16 packer plugin: Sending char 'X', code CodeX, shift false
2020/12/09 15:24:16 packer plugin: Sending char 'X', code CodeX, shift false
2020/12/09 15:24:16 packer plugin: Sending char 'X', code CodeX, shift true
2020/12/09 15:24:16 packer plugin: Sending char 'X', code CodeX, shift true
2020/12/09 15:24:17 packer plugin: Sending char 'X', code CodeX, shift true
2020/12/09 15:24:17 packer plugin: Sending char 'X', code CodeX, shift false
2020/12/09 15:24:17 packer plugin: Sending char 'X', code CodeX, shift false
2020/12/09 15:24:17 packer plugin: Sending char 'X', code CodeX, shift true
2020/12/09 15:24:17 packer plugin: Sending char 'X', code CodeX, shift false
2020/12/09 15:24:17 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:24:18 packer plugin: Sending char 'l', code CodeL, shift false
2020/12/09 15:24:18 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:24:18 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:24:18 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:24:18 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:18 packer plugin: Sending char 'v', code CodeV, shift false
2020/12/09 15:24:18 packer plugin: Sending char 't', code CodeT, shift false
2020/12/09 15:24:19 packer plugin: Sending char 'y', code CodeY, shift false
2020/12/09 15:24:19 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:19 packer plugin: Sending char '0', code Code0, shift false
2020/12/09 15:24:19 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:19 packer plugin: Sending char '1', code Code1, shift false
2020/12/09 15:24:19 packer plugin: Sending char '5', code Code5, shift false
2020/12/09 15:24:20 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:24:20 packer plugin: Sending char 'l', code CodeL, shift false
2020/12/09 15:24:20 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:24:20 packer plugin: Sending char 'g', code CodeG, shift false
2020/12/09 15:24:20 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:24:20 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:24:21 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:21 packer plugin: Sending char 'l', code CodeL, shift false
2020/12/09 15:24:21 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:24:21 packer plugin: Sending char 'c', code CodeC, shift false
2020/12/09 15:24:21 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:24:21 packer plugin: Sending char 'l', code CodeL, shift false
2020/12/09 15:24:22 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:24:22 packer plugin: Sending char 't', code CodeT, shift false
2020/12/09 15:24:22 packer plugin: Sending char 'r', code CodeR, shift false
2020/12/09 15:24:22 packer plugin: Sending char 'a', code CodeA, shift false
2020/12/09 15:24:22 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:24:22 packer plugin: Sending char 's', code CodeS, shift false
2020/12/09 15:24:22 packer plugin: Sending char 'p', code CodeP, shift false
2020/12/09 15:24:22 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:24:23 packer plugin: Sending char 'r', code CodeR, shift false
2020/12/09 15:24:23 packer plugin: Sending char 't', code CodeT, shift false
2020/12/09 15:24:23 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:23 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:24:23 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:24:23 packer plugin: Sending char 'p', code CodeP, shift false
2020/12/09 15:24:23 packer plugin: Sending char 'u', code CodeU, shift false
2020/12/09 15:24:23 packer plugin: Sending char 't', code CodeT, shift false
2020/12/09 15:24:24 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:24 packer plugin: Sending char 's', code CodeS, shift false
2020/12/09 15:24:24 packer plugin: Sending char 's', code CodeS, shift false
2020/12/09 15:24:24 packer plugin: Sending char 'h', code CodeH, shift false
2020/12/09 15:24:24 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:24:24 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:24:24 packer plugin: Sending char 'x', code CodeX, shift false
2020/12/09 15:24:24 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:24:25 packer plugin: Sending char 't', code CodeT, shift false
2020/12/09 15:24:25 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:24:25 packer plugin: Sending char 'd', code CodeD, shift false
2020/12/09 15:24:25 packer plugin: Sending char 'o', code CodeO, shift false
2020/12/09 15:24:25 packer plugin: Sending char ' ', code CodeSpacebar, shift false
2020/12/09 15:24:25 packer plugin: Sending char 'w', code CodeW, shift false
2020/12/09 15:24:25 packer plugin: Sending char 'r', code CodeR, shift false
2020/12/09 15:24:25 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:24:26 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:24:26 packer plugin: Sending char 'n', code CodeN, shift false
2020/12/09 15:24:26 packer plugin: Sending char 'd', code CodeD, shift false
2020/12/09 15:24:26 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:24:26 packer plugin: Sending char 'e', code CodeE, shift false
2020/12/09 15:24:26 packer plugin: Sending char 'x', code CodeX, shift false
2020/12/09 15:24:26 packer plugin: Sending char 'i', code CodeI, shift false
2020/12/09 15:24:26 packer plugin: Sending char 't', code CodeT, shift false
2020/12/09 15:24:27 packer plugin: Special code '<enter>' found, replacing with: CodeReturnEnter
2020/12/09 15:24:27 packer plugin: [INFO] Waiting for IP, up to total timeout: 30m0s, settle timeout: 1m0s
2020/12/09 15:24:27 ui: [1;32m==> vsphere-iso: Waiting for IP...[0m
2020/12/09 15:54:27 ui error: [1;31m==> vsphere-iso: Timeout waiting for IP.[0m
2020/12/09 15:54:27 ui: [1;32m==> vsphere-iso: Clear boot order...[0m
2020/12/09 15:54:27 ui: [1;32m==> vsphere-iso: Power off VM...[0m
2020/12/09 15:54:27 ui: [1;32m==> vsphere-iso: Destroying VM...[0m
2020/12/09 15:54:28 [INFO] (telemetry) ending vsphere-iso
2020/12/09 15:54:28 ui error: [1;31mBuild 'vsphere-iso' errored after 36 minutes 13 seconds: Timeout waiting for IP.[0m
2020/12/09 15:54:28 ui: 
==> Wait completed after 36 minutes 13 seconds
2020/12/09 15:54:28 machine readable: error-count []string{"1"}
2020/12/09 15:54:28 ui error: 
==> Some builds didn't complete successfully and had errors:
2020/12/09 15:54:28 machine readable: vsphere-iso,error []string{"Timeout waiting for IP."}
2020/12/09 15:54:28 ui error: --> vsphere-iso: Timeout waiting for IP.
2020/12/09 15:54:28 ui: 
==> Builds finished but no artifacts were created.
2020/12/09 15:54:28 [INFO] (telemetry) Finalizing.
2020/12/09 15:54:30 waiting for all plugin processes to complete...
2020/12/09 15:54:30 templates/packer/packer: plugin process exited
```
</details>

**Comment by timblaktu** [comment]
@bruteForceAttack we're experiencing what could be a similar issue. We're using a virtualbox-iso builder, and we're building Debian 10 images on a Debian 10 host, but we're also using 1.6.5 and are also getting timeouts waiting for VM/ssh. In our case, we know the reason for the timeout is because the preseeded installer is hung because it's got a hostname with bad characters in it that appear to be corruption of the `vm_name` we defined in our template including a user variable (something you're doing as well).

The problem seems to go away when we just use a constant vm_name..

In our case, we're defining:

```
  "variables": {
    "version": ""
  },
```
and in our virtualbox-iso builder, we declare:

```
      "vm_name": "packer-debian-10-amd64-{{user `version`}}",
```

In our case, we have tried passing in version using -var and -vars-file packer args and both ways, we intermittently get the aforementioned problem where the preseeded installer is hung bc the hostname (which is derived from the vm_name somehow) has invalid chars in it (for a hostname). 

The bad hostname in the installer never has the "packer-debian-10-amd64-" part corrupted, it's always the suffix that we're substituting in from user variable "version" which is corrupted, 

e.g. if version = "bugfix-SWOPS-736-Fix-for-1a71557-4"

we might then get a hostname that looks like this:

 packer-debian-10-amd64-feature-SWOPS_#^_BUILDING_##%(BB(!

In the above example, we are passing version in a packer -vars-file in a sh step in a Jenkins pipeline, which "sanitizes" a string based on branch name, commit sha, and build number, to be used as the vm_name suffix, as follows:

```
                    sh """echo '{"version": "${GIT_BRANCH.replaceAll('/','').take(24)}-${GIT_COMMIT.take(7)}-${BUILD_NUMBER}"}'"""
                    sh """echo '{"version": "${GIT_BRANCH.replaceAll('/','-').take(24)}-${GIT_COMMIT.take(7)}-${BUILD_NUMBER}"}' > vars.json"""
                    sh """cat vars.json"""
                    sh 'packer build -color=false -force -var-file=vars.json packer.json'
```

The echo steps indicate what looks like a properly-sanitized string that could be used as a hostname:

```
15:03:31  [Pipeline] sh
15:03:31  + echo {"version": "bugfix-SWOPS-736-Fix-for-1a71557-4"}
15:03:31  {"version": "bugfix-SWOPS-736-Fix-for-1a71557-4"}
15:03:31  [Pipeline] sh
15:03:31  + echo {"version": "bugfix-SWOPS-736-Fix-for-1a71557-4"}
15:03:31  [Pipeline] sh
15:03:31  + cat vars.json
15:03:31  {"version": "bugfix-SWOPS-736-Fix-for-1a71557-4"}
15:03:31  [Pipeline] sh
15:03:32  + packer build -color=false -force -var-file=vars.json packer.json
```

And then later, we see in the packer console that this value is used correctly:

15:03:35      virtualbox-iso: Executing: modifyvm packer-debian-10-amd64-bugfix-SWOPS-736-Fix-for-1a71557-4 --memory 4096

But then when we time out waiting for the machine, we notice that it has the bogus hostname, with the suffix portion-only corrupted. 

Flummoxed. Trying to track down the conditions in which this corruption occurs, and will report back, or perhaps create a new issue here.

**Comment by hfenner** [comment]
I believe I have the same issue with waiting for an IP.  I'm attempting to build a vyos system and the system is clearly getting an IP, but Packer is not reading it.   We're attempting to build a VyOS IP with the `vsphere-iso` builder using extensive boot commands.

**Error:**
```
2021/04/13 11:16:50 packer-builder-vsphere-iso plugin: [INFO] Waiting for IP, up to total timeout: 30m0s, settle timeout: 5s
==> vyos_vsphere_iso: Waiting for IP...
==> vyos_vsphere_iso: Timeout waiting for IP.
==> vyos_vsphere_iso: Power off VM...
==> vyos_vsphere_iso: Destroying VM...
2021/04/13 11:46:51 [INFO] (telemetry) ending vsphere-iso
==> Wait completed after 39 minutes 4 seconds
2021/04/13 11:46:51 machine readable: error-count []string{"1"}
```

**Screenshot of Packer hanging:**
![Hung](https://user-images.githubusercontent.com/63067131/114580390-cfa3d400-9c4c-11eb-99f1-78f68179c295.png)

**Screenshot of IP acquired in vSphere GUI:**
![icanhazip](https://user-images.githubusercontent.com/63067131/114580281-b733b980-9c4c-11eb-9561-e8b80c557a61.png)

**Screenshot of govc returning IP:**
```
[packer]$ govc vm.info vyos
Name:           vyos
  Path:         /Datacenter/vm/Templates/vyos
  UUID:         421ebb6d-8ab3-f2f8-b6b8-dda2c4696049
  Guest name:   Ubuntu Linux (64-bit)
  Memory:       4096MB
  CPU:          2 vCPU(s)
  Power state:  poweredOn
  Boot time:    2021-04-13 15:08:12.459938 +0000 UTC
  IP address:   192.168.173.112
  Host:         10.15.15.206
```

**Comment by bruteForceAttack** [comment]
@hfenner - Seems to be very similar problem however I did some tests with those versions of vyos rolling release:

- vyos-1.3-rolling-202002190654-amd64.iso
- vyos-1.4-rolling-202102221801-amd64.iso 

and everything seems to be just fine though. Tested with Packer 1.7.0.

**Comment by hfenner** [comment]
I'm running Packer 1.7.2 and I will try your specific VyOS and Packer releases.

**Comment by hfenner** [comment]
I couldn't acquire your specific VyOS builds but I did attempt the build with the latest 1.4 iso/Packer 1.7.2 and had no luck.  Would you be kind enough to share your Packer file? I'm targeting vSphere 6.7 if it matters.

I will attempt with Packer 1.7.0 tomorrow for good measure.

**Comment by hfenner** [comment]
1.7.0 does not seem to work for me with vyos-1.4-rolling-202102221801-amd64.iso (https://downloads.vyos.io/rolling/current/amd64/vyos-rolling-latest.iso).

I'm at a little bit of a loss on why it works for you but not for me.

This is the Packer file I'm using.  Right now I'm using the JSON format; I did try to the HCL2 format at one point, but reverted to JSON to make it easier to test older binaries.

`vyos.json`
```
{                                                                                                                                                                                                                                                                                                                             
  "builders": [                                                                                                                                                                                                                                                                                                               
    {                                                                                                                                                                                                                                                                                                                         
      "type": "vsphere-iso",                                                                                                                                                                                                                                                                                                  
      "name": "vyos_vsphere_iso",                                                                                                                                                                                                                                                                                             
      "convert_to_template": true,                                                                                                                                                                                                                                                                                            
      "CPUs": 2,                                                                                                                                                                                                                                                                                                              
      "RAM": 4096,                                                                                                                                                                                                                                                                                                            
      "RAM_reserve_all": true,                                                                                                                                                                                                                                                                                                
      "username": "<insertusernamehere>",                                                                                                                                                                                                                                                                              
      "password": "<insertpasswordhere>",                                                                                                                                                                                                                                                                                         
      "vcenter_server": "10.15.15.200",                                                                                                                                                                                                                                                                                       
      "cluster": "206",                                                                                                                                                                                                                                                                                                       
      "datacenter": "Datacenter",                                                                                                                                                                                                                                                                                             
      "datastore": "DATASTORE01",                                                                                                                                                                                                                                                                                             
      "folder": "Templates",                                                                                                                                                                                                                                                                                                  
      "disk_controller_type": "pvscsi",                                                                                                                                                                                                                                                                                       
      "firmware": "bios",                                                                                                                                                                                                                                                                                                     
      "boot_wait": "5s",                                                                                                                                                                                                                                                                                                      
      "boot_order": "disk,cdrom,floppy",                                                                                                                                                                                                                                                                                      
      "boot_command": [                                                                                                                                                                                                                                                                                                       
        "<enter>",                                                                                                                                                                                                                                                                                                            
        "<wait60>",                                                                                                                                                                                                                                                                                                           
        "vyos<enter><wait5>",                                                                                                                                                                                                                                                                                                 
        "vyos<wait><enter><wait10>",                                                                                                                                                                                                                                                                                          
        "install image<enter><wait5>",                                                                                                                                                                                                                                                                                        
        "<enter><wait5>",                                                                                                                                                                                                                                                                                                     
        "<enter><wait5>",                                                                                                                                                                                                                                                                                                     
        "<enter><wait5>",                                                                                                                                                                                                                                                                                                     
        "Yes<enter><wait5>",                                                                                                                                                                                                                                                                                                  
        "<enter><wait10>",                                                                                                                                                                                                                                                                                                    
        "<enter><wait5>",                                                                                                                                                                                                                                                                                                     
        "<enter><wait5>",                                                                                                                                                                                                                                                                                                     
        "vyos<enter><wait5>",                                                                                                                                                                                                                                                                                                 
        "vyos<enter><wait10>",                                                                                                                                                                                                                                                                                                
        "<enter><wait20>",                                                                                                                                                                                                                                                                                                    
        "reboot<enter><wait5>",                                                                                                                                                                                                                                                                                               
        "Yes<enter><wait3m>",                                                                                                                                                                                                                                                                                                 
        "vyos<enter><wait5>",                                                                                                                                                                                                                                                                                                 
        "vyos<enter><wait10>",                                                                                                                                                                                                                                                                                                
        "configure<enter><wait5>",                                                                                                                                                                                                                                                                                            
        "set interface ethernet eth0 address dhcp<enter><wait5>",                                                                                                                                                                                                                                                             
        "set service ssh<enter><wait5>",                                                                                                                                                                                                                                                                                      
        "commit<enter><wait5>",                                                                                                                                                                                                                                                                                               
        "save<enter><wait5>",                                                                                                                                                                                                                                                                                                 
        "delete interface ethernet eth0 hw-id<enter><wait5>",                                                                                                                                                                                                                                                                 
        "commit<enter><wait5>",                                                                                                                                                                                                                                                                                               
        "save<enter><wait5>",                                                                                                                                                                                                                                                                                                 
        "exit<enter><wait5>",                                                                                                                                                                                                                                                                                                 
        "sudo bash -c 'echo deb https://deb.debian.org/debian buster main contrib non-free > /etc/apt/sources.list.d/buster.list'<enter><wait5>",                                                                                                                                                                             
        "sudo bash -c 'echo nameserver 1.1.1.1 >> /etc/resolv.conf'<enter><wait5>",                                                                                                                                                                                                                                           
        "sudo apt update<enter><wait5>",                                                                                                                                                                                                                                                                                      
        "sudo apt install -y open-vm-tools<enter><wait40>"                                                                                                                                                                                                                                                                    
      ],                                                                                                                                                                                                                                                                                                                      
      "guest_os_type": "ubuntu64Guest",                                                                                                                                                                                                                                                                                       
      "insecure_connection": "true",                                                                                                                                                                                                                                                                                          
      "iso_checksum": "none",                                                                                                                                                                                                                                                                                                 
      "ssh_username": "vyos",                                                                                                                                                                                                                                                                                                 
      "ssh_password": "vyos",                                                                                                                                                                                                                                                                                                 
      "ip_wait_timeout": "5m",                                                                                                                                                                                                                                                                                                
      "iso_urls": [                                                                                                                                                                                                                                                                                                           
        "https://downloads.vyos.io/rolling/current/amd64/vyos-rolling-latest.iso"                                                                                                                                                                                                                                             
      ],                                                                                                                                                                                                                                                                                                                      
      "network_adapters": [                                                                                                                                                                                                                                                                                                   
        {                                                                                                                                                                                                                                                                                                                     
          "network": "1701 - OFFNET",                                                                                                                                                                                                                                                                                         
          "mac_address": "00:50:56:AA:CC:BB",                                                                                                                                                                                                                                                                                 
          "network_card": "vmxnet3"                                                                                                                                                                                                                                                                                           
        }                                                                                                                                                                                                                                                                                                                     
      ],                                                                                                                                                                                                                                                                                                                      
      "vm_name": "vyos",
      "communicator": "ssh",
      "notes": "SSH Username vyos\nSSH Password vyos",
      "storage": [
        {
          "disk_size": 16384,
          "disk_thin_provisioned": true
        }
      ]
    }
  ]
}

**Comment by hfenner** [comment]
After further experimentation, I've been able to determine that 1.6.0 works for the above build but 1.6.1 does not.  I do think this has something to do with the same issue @bruteForceAttack identified.

**Comment by hfenner** [comment]
Success! When I removed the static MAC address assignment, things worked! I think there's something about parsing a static MAC that is broken in this change.
